server/graphql/general/src/mutations/display_settings.rs

```diff
@@ -0,0 +1,75 @@
+use async_graphql::*;
+
+use graphql_core::{standard_graphql_error::validate_auth, ContextExt};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    display_settings_service,
+};
+#[derive(InputObject)]
+pub struct DisplaySettingsInput {
+    pub custom_logo: Option<String>,
+    pub custom_theme: Option<String>,
+}
+
+#[derive(SimpleObject)]
+pub struct UpdateResult {
```

Comment:
originally I returned the display settings as the response - but the svg files that we have can get quite large, and since we aren't using the result I thought it best not to slow down the response for users on poor connections

Reply:
Hmm, updating settings is rarely done, personally I don't think it's worth the extra code, and are we not returning this svg everytime page is refreshed ?

Reply:
Actually thinking this through again, if we want to minimise traffic, i think we should do this:

`front end`
top level provider (one of the first queries that runs on first app load or refresh)
* get stored `hash` of logo and customer_theme from local storage
* use that `hash` as a query paremeter for display settings
* back end would return either new hash and settings (if it's different) otherwise would return `null`
* front end stores new hash if settings change and the new value for the settings, otherwise loads existing value for the settings from localStorage
`back end`
* When storing settings hash them and store as extra key_value
* When queried expects hash of the settings, and compares with stored hash and either return empty result or settings that's been changed (value and hash)

So in this case API would be:

```gql
displaySettings2(input: DisplaySettingsInput2!): DisplaySettingsNode2!

input DisplaySettingsInput2 {
	"""
	Hash of existing custom logo
	"""
	customLogoHash: String
	"""
	Hash of existing custom theme
	"""
	customThemeHash: String
}

type DisplaySettingsNode2 {
	"""
	Custom logo, null if not defined or if hash in query is unchanged
	"""
	customLog: DisplaySettingsNode
	"""
	Custom theme, null if not defined or if hash in query is unchanged
	"""
	customTheme: DisplaySettingNode
}

type DisplaySettingNode {
	value: String!
	hash: String!
}

```
rust definition:
```rust
 
    pub async fn display_settings2(
        &self,
        _input: DisplaySettingsInput2,
    ) -> Result<DisplaySettingsNode2> {
        todo!()
    }
}

#[derive(SimpleObject)]
pub struct DisplaySettingNode {
    pub value: String,
    pub hash: String,
}

#[derive(SimpleObject)]
pub struct DisplaySettingsNode2 {
    /// Custom logo, null if not defined or if hash in query is unchanged
    pub custom_log: Option<DisplaySettingsNode>,
    /// Custom theme, null if not defined or if hash in query is unchanged
    pub custom_theme: Option<DisplaySettingNode>,
}

#[derive(InputObject)]
pub struct DisplaySettingsInput2 {
    /// Hash of existing custom logo
    pub custom_logo_hash: Option<String>,
    /// Hash of existing custom theme
    pub custom_theme_hash: Option<String>,
}
```